### ISSUE TITLE
Slow board with lots of swimlanes/lists

### ISSUE THREAD

**Issue by GuillaumeV-cemea** [wekan/wekan#5208]
### Server Setup Information

* Did you test in newest Wekan?: Only in 7.17
* Did you configure root-url correctly so Wekan cards open correctly (see https://github.com/wekan/wekan/wiki/Settings)? yes
* Operating System: Debian
* Deployment Method (Snap/Docker/Sandstorm/bundle/source): Docker
* Http frontend if any (Caddy, Nginx, Apache, see config examples from Wekan GitHub wiki first): nginx
* Node.js Version: 14.21.4
* MongoDB Version: 6.0.11
* What webbrowser version are you using (Wekan should work on all modern browsers that support Javascript)? Firefox latest, Chromium latest

### Problem description

After upgrading from wekan 6.09 to wekan 7.17, some of our boards are painfully slow to load/use. I have the following board (I took the stats from an export and counted the number of elements, so it includes both normal elements and archived elements) : 
  * 21 lists
  * 22 swimlanes
  * 467 cards
  * no attachments
  * 10 members
  * 1156 activities
 
Wekan is running inside docker,  and the VM has 6 vCPU (so 3 core/6 thread from an AMD EPYC 7451), with 6 GB of RAM, the disk are fast nvme disks. The only changes I made to my docker compose is root_url/email_url, no performance changes.

When I load this board, it takes around 11 seconds (!) to fully load (from entering the URL in my address bar and pressing Enter to the board is functionnal).

I have tested in Firefox latest and chromium latest, no extensions, the results are similar. It happens on multiple computer, on Linux / Windows / Mac, the strongest CPU beeing a i5-1135G7 (should be more than enough).

During this load, I can see in about:performance in Firefox (or in chromium task manager), I can see RAM usage spiking to 700MB, and CPU usage going to 100% (sometimes a bit over but it's basically 100%), the CPU usage staying at 100% for around 20 seconds total (so I'm thinking the board is not totally loaded and doing some work after it visually looks like it's totally loaded). After that, memory usage goes down to 400 MB, and CPU usage is practically 0.

Switching from this board  to another page (another board, or admin settings) is also painfully slow : switching to admin settings is taking me 25seconds with the same behaviour (100% CPU, RAM usage spike)
 
 For reference, I have a board with 13 lists, 4 swimlanes, 741 cards, no attachments, 5 members and 2491 activities (so not a small board either), and it takes around 2 seconds to load.

What can I do to help solve this ? I can provide in private an export of the board (it contains some private data and considering the amount of cards, anonymizing it to show it in public would be difficult) if it helps, and try to do some tests to try to find which version(s) introduced this lag issue.

#### Reproduction Steps

Load a slow board, wait an eternity for it to load.

**Comment by mfilser** [comment]
@GuillaumeV-cemea
In total with the latest version 7.17 one board loads within 2 seconds and the other one takes 11 seconds, right?

**Comment by GuillaumeV-cemea** [comment]
@mfilser Yes, exactly !

**Comment by mfilser** [comment]
I see.The one board with 2s has 4 swimlane and the 11s board 22 swimlanes?

**Comment by GuillaumeV-cemea** [comment]
Yes ! I did some more experiments by creating a new instance of wekan (with the same docker parameters), version 6.09 and importing my board. The speed is actually not much slower (I'm at around 10-12sec too), however it "feels" slower because in v7.17, it "halfs load" a board where the lists are very narrow, you already see the cards but you cannot do anything.


  * Removing all archives get me down to 14 swimlanes, 18 lists, 376 cards (there is also 22 labels on the board, I forgot about this). It's a bit faster (9sec instead of 11 sec)
  * Removing all labels has no effect on load speed
  * Removing (archiving + removing from archive) to get to 5 swimlanes gives a noticeable speed boost (down to 5 seconds to load the board)
  * Removing (archiving + removing from archive) to get to 6 lists gives the same speed boost (5 seconds to load the board)
  * 5 swimlanes + 6 lists is around 3.5 seconds to load

It does seems it's linked with the number of swimlanes and list, and there is no real difference from 6.09 to 7.17 (I did the tests on both)

**Comment by GuillaumeV-cemea** [comment]
Upgrading from 7.17 to 7.61 "feels" better, because I don't have the "half loading" symptom (where the lanes are loaded but very narrow, and you have to wait more for the board to be functional), but loading is still slow. 
My problematic board now has 17 swimlanes, 18 lists and 712 cards, and loading it takes around 7 seconds. Considering there is more cards/lists/swimlanes, you can see an increase in performance (twice as much cards, less time), but it's still too slow, and all other "slow" behaviour (described in the initial issue).
Restarting wekan does reduce RAM used, but I saw no difference in performance (which is logical, because it seems to be client-side as I described) regarding these number of any of the behaviour described in the original issue.

**Comment by xet7** [comment]
@GuillaumeV-cemea 

About your WeKan server:
- Does it have SSD or NVME disk?
- How much it has total and used RAM ? Check with `free -h`
- How many CPU cores your server has? Check with `nproc`
- Please upgrade to WeKan v7.64 (or newer, if available). Is there difference in loading speed?

About your laptop or desktop, where you use webbrowser with WeKan:
- What operating system version?
- What CPU it has?
- How much total RAM it has?
- Does it have SSD or NVME disk?
- What webbrowser you use? Chrome/Edge/Safari is faster than Firefox.

**Comment by xet7** [comment]
I would think, that there is difference between:

a) Running WeKan server at Apple Silicon macOS Docker, and Safari webbrowser

b) Using some slower systems, like Intel/AMD
